js/ui/windowManager.js

```diff
@@ -468,34 +468,22 @@ var WindowManager = class WindowManager {
 
         this.settings = new Settings({schema_id: 'org.cinnamon.muffin'});
 
-        const settingsState = {
```

Comment:
I think it's cleaner if we separate state from the base prototype instance. It makes it possible to make property access faster. Take for example:

```js
let something1 = this.propA;
let something2 = this.propB;
// We're asking the engine to look up multiple properties on `this`, it has to look up a key on a complex object that is possibly inheriting other prototypes, including functions, references, and primitives.
```
```
let {state} = this; // We're asking the engine to look up a key on the complex object once.
let something1 = state.propA;
let something2 = state.propB;
// We're asking the engine to look up multiple properties on a much less complex object.
```